lib/metrics/StatsClient.js

```diff
@@ -65,15 +65,30 @@ class StatsClient {
     /**
     * report/record a new request received on the server
     * @param {string} id - service identifier
-    * @param {callback} cb - callback
+    * @param {number} incr - optional param increment
+    * @param {function} cb - callback
     * @return {undefined}
     */
-    reportNewRequest(id, cb) {
+    reportNewRequest(id, incr, cb) {
         if (!this._redis) {
             return undefined;
         }
-        const callback = cb || this._noop;
+
+        let callback = this._noop;
+        let amount = 1;
+        if (typeof incr === 'function') {
+            // In case where optional `incr` is not passed, but `cb` is passed
+            callback = incr;
+        } else {
+            callback = (cb && typeof cb === 'function') ? cb : callback;
+            amount = (incr && typeof incr === 'number') ? incr : amount;
+        }
+
         const key = this._buildKey(`${id}:requests`, new Date());
+
+        if (amount !== 1) {
```

Comment:
maybe you can use `incrbyEx` in all cases, I'm sure if there's any optimization for the incr by one case the redis API will have implemented it also in `incbyEx`.